consensus/src/chained_bft/event_processor.rs

```diff
@@ -820,26 +814,51 @@ impl<T: Payload> EventProcessor<T> {
                 return;
             }
         };
-        // At this moment the new state is persisted and we can notify the clients.
-        // Multiple blocks might be committed at once: notify about all the transactions in the
-        // path from the old root to the new root.
-        for committed in blocks_to_commit {
-            if let Some(time_to_commit) = duration_since_epoch()
-                .checked_sub(Duration::from_micros(committed.timestamp_usecs()))
+
+        // update counters
```

Comment:
minor: move the counter update logic to a separate static function `update_counters_for_committed_blocks()`.